From 5f13cdce615165f5a623faa4a642ecc723202b0f Mon Sep 17 00:00:00 2001
From: Merlijn Wajer <merlijn@wizzup.org>
Date: Mon, 20 Dec 2021 22:27:03 +0100
Subject: [PATCH] ARM: dts: omap3-n900: disable thermal for now

See "Oops while booting 5.15.2 on Nokia N900" on linux-omap mailing list
---
 arch/arm/boot/dts/omap3-n900.dts | 17 +++++++++++++++++
 1 file changed, 17 insertions(+)

diff --git a/arch/arm/boot/dts/omap3-n900.dts b/arch/arm/boot/dts/omap3-n900.dts
index 1746a44e96bc..fe7c884da27c 100644
--- a/arch/arm/boot/dts/omap3-n900.dts
+++ b/arch/arm/boot/dts/omap3-n900.dts
@@ -28,6 +28,23 @@
 	status = "disabled";
 };
 
+/* Disable cpu_thermal and bandgap, they causes oopses on mmc access in OFF
+   mode at the moment. Let's just disable thermal_zones as well, although the
+   system works without it...
+*/
+&cpu_thermal {
+	status = "disabled";
+};
+
+&bandgap {
+	status = "disabled";
+};
+
+&thermal_zones {
+	status = "disabled";
+};
+
+
 / {
 	model = "Nokia N900";
 	compatible = "nokia,omap3-n900", "ti,omap3430", "ti,omap3";
